### PR TITLE
check if entity exists before accessing

### PR DESCRIPTION
**Pull request by LaserWitch**
I've run into a few panics with world access if you happen to hold on to entity values between frames. I'm not certain this is the ideal way to handle them, so feel free to suggest changes or go with an alternate implementation if desired.

### CODE REVIEW

File: bevy_script_api/src/sub_reflect.rs [hunk 1]
```diff
@@ -348,6 +348,13 @@ impl ReflectPath {
...
                 let g = world_ptr.read();
 
+                g.get_entity(*entity).ok_or_else(|| {
+                    ReflectionError::InvalidBaseReference {
+                        base: self.base.to_string(),
+                        reason: "This entity does not exist".to_owned(),
+                    }})?;
+
+
                 let ref_ = self.walk_path(comp.reflect(g.entity(*entity)).ok_or_else(|| {
```
Comment (makspll) [inline bevy_script_api/src/sub_reflect.rs:358]: Just so we don't look up the entity twice
```suggestion
                let entity_ref = g.get_entity(*entity).ok_or_else(|| {
                    ReflectionError::InvalidBaseReference {
                        base: self.base.to_string(),
                        reason: "This entity does not exist".to_owned(),
                    }})?;
                let ref_ = self.walk_path(comp.reflect(entity_ref).ok_or_else(|| {
```

Comment (makspll) [reply bevy_script_api/src/sub_reflect.rs:358]: I am assuming you were getting panics in the `g.entity` call right?

File: bevy_script_api/src/common/bevy/mod.rs [hunk 2]
```diff
@@ -169,6 +169,11 @@ impl ScriptWorld {
     ) -> Result<ScriptRef, ScriptError> {
         let mut w = self.write();
 
+        let _ = match w.get_entity(entity) {
```
Comment (makspll) [inline bevy_script_api/src/common/bevy/mod.rs:172]: can we cache the entity as in the other comment here as well?

**Review by makspll — CHANGES_REQUESTED** [review]
Great catch! Would you be okay adding similar checks where we use `w.entity` or `w.entity_mut`. Much appreciated!

File: bevy_script_api/src/sub_reflect.rs (hunk 1 above)
Comment (LaserWitch) [reply bevy_script_api/src/sub_reflect.rs:358]: Right. Good call with not doing the lookup twice, I'll incorporate that

**Comment by LaserWitch** [comment]
> Great catch! Would you be okay adding similar checks where we use `w.entity` or `w.entity_mut`. Much appreciated!

Yeah, I'll do a search and see how many other places I can add it in.

I'm honestly tempted to try to implement an option or result style construct in lua, perhaps leverage multiple value return, but I think I can experiment with that a bit further down stream for the time being.

**Comment by makspll** [comment]
> Yeah, I'll do a search and see how many other places I can add it in.
Thanks!

> I'm honestly tempted to try to implement an option or result style construct in lua, perhaps leverage multiple value return, but I think I can experiment with that a bit further down stream for the time being.

I've considered something like this in the past, although I've stuck with the mlua convention of converting `None` -> `nil` and `Error` -> Runtime Error, while creating proxy types for things like `Vec` when maintaining `pass-by-reference` behavior's.

**Comment by LaserWitch** [comment]
I've been lost in some other weeds the past week but I'll see if I can get this done this weekend.

**Comment by LaserWitch** [comment]
I have some more bases covered now. There's some other things that look like potential cases that could be checked, but I wouldn't really want to mess with them without first having a test case that is known to fail, and building those tests cases is not something I'm up for doing at the moment

**Review by makspll — DISMISSED** [review]
Looks good, if we need any more checks anywhere else we can add later. Thanks!

**Comment by LaserWitch** [comment]
Looks like I've finally made CI happy. Not sure why it says I dismissed the review above, that must have been something automatic.

**Review by makspll — APPROVED** [review]

**Comment by makspll** [comment]
Yeah it's automatic, might disable that as the message looks quite confusing.

**Pull request merged**